### DISCUSSION TITLE
Where can I find the iCub Robot Assembly Tutorial?

### DISCUSSION THREAD

**Discussion by xilihutu888** [robotology/community#477] [Q&A] [answered]
I want to spend two years assembling the iCub robot. How do I get the complete information?
Thank you very much.

**Comment by pattacini** [comment, marked as answer]
Hi @nan2854 👋🏻 

Welcome to our community ✨ 
Your goal sounds very peculiar; we hope you're equipped with all the necessary tools and parts to pursue it.

Coming down to your request, we tend to use our CAD drawings as the main reference for assembling the robot.
To this end, you could proceed this way:
- Install [Creo 7.0][1], which is our CAD software.
- Clone the repository [`cad-libraries`][2].
- Configure `cad-libraries` following these [instructions][3].
- Clone the repository [`cad-mechanics-public`][4].
- Finally, you'll be able to open up the main iCub drawing file **`icub_fbbbbeba.asm.1`** (contained in `cad-mechanics-public`) within Creo.

In case you don't have a Creo license, you may use [Creo View Express][5] as a fallback. However, you won't be able to load the commercial components. To get around this, we could then point you to one of our CAD snapshots later on.

You may also find interesting this guide we have recently made available: https://icub-tech-iit.github.io/documentation/icub_hw.

Hope this helps out.

cc @fiorisi 

[1]: https://www.ptc.com/en/products/creo/parametric
[2]: https://github.com/icub-tech-iit/cad-libraries
[3]: https://github.com/icub-tech-iit/cad-libraries/wiki/Configure-PTC-Creo-with-cad-libraries
[4]: https://github.com/icub-tech-iit/cad-mechanics-public
[5]: https://www.ptc.com/en/products/creo/creo-view/extension-express

**Reply by xilihutu888** [reply]
Thank you for your answer. What is the specific model of PC104 used by ICUB? 
Where can I download all the electronic circuit board PCB and instructions of iCub？
（An initial manual of the controllers is available from:iCub manuals repository in various formats (WARNING: some text is in Italian)This won't open
（http://www.robotcub.org/cvsweb/cvsweb.cgi/iCubPlatform/hardware/）This won't open
Where can I download the electronic circuit board MCP/MC4 PCB
Thank you for your attention and answer

**Reply by pattacini** [reply]
The latest iCub robots mount a COM EXPRESS. I don't know actually the type but you can find all the information regarding the electronics boards and wiring in the following repos:
- https://github.com/icub-tech-iit/electronics-boards-public
- https://github.com/icub-tech-iit/electronics-wiring-public

**Reply by xilihutu888** [reply]
（Execution context error: The remote host forced closure of an existing connection.）TortoiseSVN Download error（https://svn.robotology.eu/repos/iCubHardware-pub/trunk/mechanics/）Unable to download properly。
Thank you for your answer。

**Reply by pattacini** [reply]
Hi @nan2854,

Please, no longer stick to our old SVN repository, which has been deprecated in favor of the latest Git counterparts hosted on GitHub (those 2 repos I've linked above).

We are in the process of updating the broken/old links in our documentation.

**Reply by xilihutu888** [reply]
![1200px-Cards](https://user-images.githubusercontent.com/20624124/105463708-6ce51100-5ccb-11eb-9dc2-7d9b49b54523.jpg)
（https://github.com/icub-tech-iit/electronics-boards-public）I did not find the MCP electronic circuit board PCB in the picture, but only the MC4 electronic circuit board. Could you please give me a detailed answer? Thank you for your reply.

**Reply by pattacini** [reply]
I'm not responsible for the electronic and mechanical design, hence I would forward this request to @MrAndrea.

Anyway, for questions/requests that are repo-specific, please open up issues in the corresponding ticket systems as per our [guidelines][1].
In this case, the most pertinent place would be https://github.com/icub-tech-iit/electronics-boards-public/issues.

[1]: ../blob/master/.github/SUPPORT.md

**Reply by xilihutu888** [reply]
Thank you for your reply.
I am patiently waiting for @mrandrea's reply.
How to purchase all PCB of ICUB related to electronic circuit board? The PCB data in open source hardware is incomplete, and no PCB data of MCP-001, MC4-001, BLP and BLL have been found. Please send me a quotation.
![QQ截图20210123114214](https://user-images.githubusercontent.com/20624124/105567856-18e03800-5d70-11eb-8009-3d1233878ab2.png)

**Reply by pattacini** [reply]
Hi @nan2854 

As @maggia80 reported in https://github.com/icub-tech-iit/electronics-boards-public/issues/1#issuecomment-766101811, we have been using newer boards since long: MC4-plus, EMS4 and 2FOC.

You can contact our research and sales manager Andea Derito via email **andrea.derito@iit.it** ✉

**Reply by xilihutu888** [reply]
Thank you for your reply.
According to the open source data, whether the iCub robot can be independently manufactured? If not, what part of the data is still missing and how to get the data?

**Reply by pattacini** [reply]
The iCub project was thought to be open source from the beginning and from the ground up, thus including all the hardware too.
As a result, you can manufacture the robot yourself.

**Reply by xilihutu888** [reply]
Thank you for your reply.
I have checked all the open source hardware information, but I did not find the open source information about the battery backpack of iCub robot, the information about the battery backpack of BMS electronic circuit board of PCB, and the information about the electronic circuit board of HMS?
How can I obtain these materials? 
Where can I download the latest BOM?
The previous data sheet cannot be downloaded. All the things in it are old. The production of many electronic devices, such as cameras, has been stopped
Please help me.
![image](https://user-images.githubusercontent.com/20624124/105785113-1ae70880-5fb5-11eb-87fd-1f7b51469e90.png)

**Reply by pattacini** [reply]
Hi @nan2854 

As said all our public documentation regarding HW is hosted in the repositories I reported above. It may be certainly incomplete, so my recommendation is that you open up specific issues on the ticket systems of the corresponding repositories to trigger the action of the responsible persons.

cc @MrAndrea @maggia80
